### PR TITLE
Make `extern_mod_stmt_cnum` into a regular function

### PR DESCRIPTION
**Pull request by Aaron1011**
cc #83126

It's very short, so there's no need for it to go through the query
system.

### CODE REVIEW

**Comment by lcnr** [comment]
do we clear the query cache if the `extern_crate_map` changes between compilation sessions?

**Comment by Mark-Simulacrum** [comment]
I don't immediately see how it'd be different inside the query (i.e., why that would not lead to just as many problems as with this impl) -- but happy to r- this for now if you think that's warranted

**Comment by lcnr** [comment]
no, I also don't see how having a query here, so this PR seems good.

The existence of this query just feels kind of weird to me, maybe it was intended to improve incremental reuse in case `extern_crate_map` changes? But that wasn't actually the case afaict

**Comment by Aaron1011** [comment]
I think the important thing is that a `CrateId` doesn't actually store any information by itself. A query can only produce it as part of a result (which would lead to the same ICE if the `StableCrateId` changed), or call a query using it as an input (which will cause it to get re-run if the crate changes).

**Comment by michaelwoerister** [comment]
Are you all sure this is the right fix? Making something not be a query will certainly avoid triggering errors emitted by the query system's sanity checking -- but it might just turn this into a silent error. This not being a query will mean there's no dependency tracking for it.

**Comment by michaelwoerister** [comment]
@bors r- 

Sorry for barging in here, but this feels kind of fishy to me. @Aaron1011's [previous suggestion](https://github.com/rust-lang/rust/issues/83126#issuecomment-798982478) of turning this into an eval-always query seems like the safer option to me. I'll take a closer look right now.

**Comment by michaelwoerister** [comment]
Yes, I'm confident now that this should be an `eval_always` query. Like all other queries that directly read pre-populated data from a tcx field, this is an outside input into the query system (like `module_exports`, `in_scope_traits_map`, and `crate_name` for example). Turning this into a regular method will mean that we don't do any dependency tracking for it.

The fact that this has not been an `eval_always` query is the actual bug. Since the query just reads some data from the tcx, it will never have any dependencies. When `try_mark_green` tries to mark it green, it will not find any red dependencies, so it will assume that nothing has changed without ever fingerprinting the new value. 

So, to sum up: just make the query `eval_always` and the bug should be fixed.

**Comment by Aaron1011** [comment]
Closing in favor of https://github.com/rust-lang/rust/pull/83153
